### PR TITLE
Node pool changes a3u

### PR DESCRIPTION
**Pull request by annuay-google**
For node pool support:

- Extended reservations
- Supports A3 Ultra NPI

### Submission Checklist

NOTE: Community submissions can take up to 2 weeks to be reviewed.

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cluster Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: modules/compute/gke-node-pool/variables.tf [hunk 1]
```diff
@@ -386,6 +386,12 @@ variable "gke_version" {
   type        = string
 }
 
+variable "node_version" {
```
Comment (annuay-google) [inline modules/compute/gke-node-pool/variables.tf:389]: Will remove this and use `gke_version` instead. Will make the changes tomorrow IST

Comment (annuay-google) [reply modules/compute/gke-node-pool/variables.tf]: Node version is not required on prod clusters, removing this

File: modules/compute/gke-node-pool/reservation_definitions.tf [hunk 2]
```diff
@@ -14,18 +14,28 @@
  * limitations under the License.
 */
 
+# Split the input into three different lists where the details of a given reservation are at the same index across these lists. 
+locals {
+  # Specific block of an extended reservation can be targeted with exr-one/reservationBlocks/exr-one-block-1
+  # Data source needs to be queried with the reservation name only. So, we extract the reservation name
+  input_reservation_names    = [for r in try(var.reservation_affinity.specific_reservations, []) : split("/", r.name)[0]]
```
Comment (tpdownes) [inline modules/compute/gke-node-pool/reservation_definitions.tf:21]: nit: I think you would be better off with default values for specific_reservations of `[]` and validation blocks that protect against it being set to null. e.g. in the variables.tf file for guest_accelerator.

Then you can avoid the `try` function here and the intent becomes clearer. I would create a task to look at this for future work.

```
  validation {
    condition     = alltrue([for ga in var.guest_accelerator : ga.gpu_driver_installation_config != null])
    error_message = "var.guest_accelerator[*].gpu_driver_installation_config must not be null; leave unset to enable GKE to select default GPU driver installation"
  }
```

**Review by tpdownes — COMMENTED** [review]
1. A `nit` for future work.
2. I've triggered all GKE tests to confirm they pass.

File: modules/compute/gke-node-pool/reservation_definitions.tf (hunk 2 above)
Comment (annuay-google) [reply modules/compute/gke-node-pool/reservation_definitions.tf:21]: Done, have tagged you in the task as reporter

**Pull request merged**